backend/src/middleware/notifications/posts-in-groups.spec.ts

```diff
@@ -0,0 +1,395 @@
+          groupId: 'g-1',
+        },
+      })
+    })
+
+    it('sends NO notification to the author of the post', async () => {
+      await expect(
+        query({
+          query: notificationQuery,
+          variables: {
+            read: false,
+          },
+        }),
+      ).resolves.toMatchObject({
+        data: {
+          notifications: [],
+        },
+        errors: undefined,
+      })
+    })
+
+    it('sends NO notification to the pending group member', async () => {
+      authenticatedUser = await pendingMember.toJson()
+      await expect(
+        query({
+          query: notificationQuery,
+          variables: {
+            read: false,
+          },
+        }),
+      ).resolves.toMatchObject({
+        data: {
+          notifications: [],
+        },
+        errors: undefined,
+      })
+    })
+
+    it('sends notification to the group member', async () => {
+      authenticatedUser = await groupMember.toJson()
+      await expect(
+        query({
+          query: notificationQuery,
+          variables: {
+            read: false,
+          },
+        }),
+      ).resolves.toMatchObject({
+        data: {
+          notifications: [
+            {
+              from: {
+                __typename: 'Post',
+                id: 'post',
+              },
+              read: false,
+              reason: 'post_in_group',
+            },
+          ],
+        },
+        errors: undefined,
+      })
+    })
+
+    it('sends one email', () => {
+      expect(sendMailMock).toHaveBeenCalledTimes(1)
+    })
+
+    describe('group member mutes group', () => {
+      it('sets the muted status correctly', async () => {
+        authenticatedUser = await groupMember.toJson()
+        await expect(
+          mutate({
+            mutation: muteGroupMutation,
+            variables: {
+              id: 'g-1',
+            },
+          }),
+        ).resolves.toMatchObject({
+          data: {
+            muteGroup: {
+              isMutedByMe: true,
+            },
+          },
+          errors: undefined,
+        })
+      })
+
+      it('sends no notification when another post is posted', async () => {
+        authenticatedUser = await groupMember.toJson()
+        await markAllAsRead()
+        authenticatedUser = await postAuthor.toJson()
+        await mutate({
+          mutation: createPostMutation,
+          variables: {
+            id: 'post-1',
+            title: 'This is another  post in the group',
+            content: 'This is the content of another post in the group',
+            groupId: 'g-1',
+          },
+        })
+        authenticatedUser = await groupMember.toJson()
+        await expect(
+          query({
+            query: notificationQuery,
+            variables: {
+              read: false,
+            },
+          }),
+        ).resolves.toMatchObject({
+          data: {
+            notifications: [],
+          },
+          errors: undefined,
+        })
```

Comment:
expect no email

Reply:
Of course there is no email, when there is no notification. So why testing it

Reply:
So we are sure this mechanic is working properly